core/src/layers/logging.rs

```diff
@@ -105,8 +106,14 @@ impl LoggingLayer {
     /// For example: accessor returns NotFound.
     ///
     /// `None` means disable the log for error.
-    pub fn with_error_level(mut self, level: Option<Level>) -> Self {
-        self.error_level = level;
+    /// Rollback to default value (`Level::Warn`) when parsing fails.
+    pub fn with_error_level(mut self, level: Option<&str>) -> Self {
+        if let Some(level) = level {
+            let level = Level::from_str(level).map_or(Some(Level::Warn), Some);
```

Comment:
Fallback to default is confusing, let's return an error here instead. This is already an API change.